src/shared/services/schema-fixer.service.ts

```diff
@@ -30,17 +30,86 @@ export class SchemaFixerService {
   constructor(private jsonUtilService: JsonUtilService) { }
 
   /**
-   * Enriches schema with options coming from configuration
+   * Fixes schema to be in a format that expected by json-editor
    * 
-   * @param {Object} schema - schema for the record
+   * @param {Object} schema - json schema
    * @param {Object} config - schema specific options
-   * @return {Object} - fixed record
+   * @return {Object} - fixed schema
    */
   fixSchema(schema: Object, config: Object): Object {
+
+    this.enrichSchemaWithConfig(schema, config);
+    schema = this.fixRecursively(schema);
+    return schema;
+  }
+
+  /**
+   * Enriches given schema with given configuration objects
+   * puts config into correct places in schema.
+   * 
+   * @param {Object} schema - json schema
+   * @param {Object} config - schema specific options
+   */
+  private enrichSchemaWithConfig(schema: Object, config: Object) {
     Object.keys(config).forEach(field => {
       let schemaField = this.jsonUtilService.getValueInPath(schema['properties'], field);
       Object.assign(schemaField, config[field]);
     });
+  }
+  /**
+   * Applies all fixes to schema recursively
+   */
+  private fixRecursively(schema: Object): Object {
+    if (schema['anyOf']) {
+      schema = this.fixAnyOf(schema);
+    }
+    // schema fixes must be done above
+
+    // recursively call for deeper parts of schema
+    if (schema['properties']) {
+      Object.keys(schema['properties'])
+        .forEach(prop => {
+          schema['properties'][prop] = this.fixRecursively(schema['properties'][prop]);
+        });
+    } else if (schema['items']) {
+      schema['items'] = this.fixRecursively(schema['items']);
+    }
     return schema;
   }
+
+  /**
+   * Fixes anyOf schemas with exactly same property structure
+   * it merges all enum fields in anyOf elements
+   */
+  private fixAnyOf(schema: Object): Object {
+    let anyOf: Array<Object> = schema['anyOf'];
+    // find enum types
+    let anyOfProps = anyOf[0]['properties'];
+    let enumPropNames = Object.keys(anyOfProps)
+      .filter(prop => anyOfProps[prop]['enum']);
+    // concat enum values for each enum property
+    let enums = {};
+    anyOf.forEach(anyOfElement => {
+      enumPropNames.forEach(propName => {
+        if (!enums[propName]) {
+          enums[propName] = [];
+        }
+        enums[propName] = enums[propName].concat(anyOfElement['properties'][propName]['enum']);
+      });
+    });
+    let fixedSchema = anyOf[0];
+    // shallow cleaning of format and pattern
+    Object.keys(fixedSchema['properties'])
+      .forEach(prop => {
+        delete fixedSchema['properties'][prop]['format'];
```

Comment:
Shall we delete specific ones or filter the ones we would like to keep (like `value`) and delete the rest? I am saying in case there is some other property that could have unexpected side effects.